src/main/java/com/hubspot/jinjava/lib/tag/SetTag.java

```diff
@@ -126,6 +129,43 @@ public String interpret(TagNode tagNode, JinjavaInterpreter interpreter) {
     return "";
   }
 
+  public String interpretBlockSet(TagNode tagNode, JinjavaInterpreter interpreter) {
+    int filterPos = tagNode.getHelpers().indexOf('|');
+    String var = tagNode.getHelpers().trim();
+    if (filterPos >= 0) {
+      var = tagNode.getHelpers().substring(0, filterPos).trim();
+    }
+    StringBuilder sb = new StringBuilder();
+    for (Node child : tagNode.getChildren()) {
+      sb.append(child.render(interpreter));
+    }
+    String[] varAsArray = new String[] { var };
+    try {
+      executeSet(
+        (TagToken) tagNode.getMaster(),
+        interpreter,
+        varAsArray,
+        Collections.singletonList(sb.toString()),
+        false
+      );
+      Object finalVal = interpreter.resolveELExpression(
+        tagNode.getHelpers().trim(),
+        tagNode.getMaster().getLineNumber()
+      );
+      executeSet(
+        (TagToken) tagNode.getMaster(),
+        interpreter,
+        varAsArray,
+        Collections.singletonList(finalVal),
+        false
+      );
```

Comment:
Why is `executeSet` executed twice? It looks like this would allow for variable variable names, but I wasn't aware that was a supported/necessary feature of jinjava. Is there another reason?

Reply:
Good question, I should probably clarify this in the code, but Jinja supports the usage of filters within set blocks like:
```
{% set foo | upper %}
Some text
{% endset %}
```
`foo == 'SOME TEXT'`.
It is easy to do this by first saving `'Some text'` to foo, and then evaluating `foo | upper` and setting the result of that expression to `foo`